pyriemann/utils/tangentspace.py

```diff
@@ -318,6 +360,49 @@ def log_map_riemann(X, Cref, C12=False):
     return X_new
 
 
+def log_map_wasserstein(X, Cref):
+    r"""Project matrices in tangent space by Wasserstein logarithmic map.
+
+    The projection of a matrix :math:`\mathbf{X}` from SPD/HPD manifold
+    to tangent space by the Wasserstein logarithmic map
+    according to a SPD/HPD reference matrix :math:`\mathbf{C}_\text{ref}` is
+    given in [1]_:
+
+    .. math::
+        \mathbf{X}_\text{new} = (\mathbf{X}\mathbf{C}_\text{ref})^{1/2} +
+        (\mathbf{C}_\text{ref}\mathbf{X})^{1/2} - 2\mathbf{C}_\text{ref}
+
+    Parameters
+    ----------
+    X : ndarray, shape (..., n, n)
+        Matrices in SPD/HPD manifold.
+    Cref : ndarray, shape (n, n)
+        Reference SPD/HPD matrix.
+
+    Returns
+    -------
+    X_new : ndarray, shape (..., n, n)
+        Matrices projected in tangent space.
+
+    Notes
+    -----
+    .. versionadded:: 0.8
+
+    References
+    ----------
+    .. [1] `Wasserstein Riemannian geometry of Gaussian densities
+        <https://link.springer.com/article/10.1007/s41884-018-0014-4>`_
+        L. Malagò, L. Montrucchio, G. Pistone. Information Geometry, 2018, 1,
+        pp. 137–179.
+    """
+    _check_dimensions(X, Cref)
+    P12 = sqrtm(Cref)
+    P12inv = invsqrtm(Cref)
+    sqrt_bracket = sqrtm(P12 @ X @ P12)
+    tmp = P12inv @ sqrt_bracket @ P12
+    return tmp + tmp.conj().swapaxes(-2, -1) - 2 * Cref
```

Comment:
Following equation, it should be
```suggestion
    return sqrtm(X @ Cref) + sqrtm(Cref @ X) - 2 * Cref
```

Reply:
X @ Cref is not symmetric, so sqrtm does not work.  
we have (X @ Cref)^0.5 = Cref^-0.5@(Cref^0.5@X@Cref^0.5)^0.5@Cref^0.5

so the part in the bracket is symmetric, thats why its wirtten like this.